React/Modules/RCTDevSettings.mm

```diff
@@ -0,0 +1,572 @@
+    // needed to prevent overriding a custom executor with the default if a
+    // custom executor has been set directly on the bridge
+    if (executorClass == Nil &&
+        _bridge.executorClass != objc_lookUpClass("RCTWebSocketExecutor")) {
+      return;
+    }
+    
+    _bridge.executorClass = executorClass;
+    [_bridge reload];
+  }
+}
+
+#pragma mark - internal
+
+/**
+ *  Query the data source for all possible settings and make sure we're doing the right
+ *  thing for the state of each setting.
+ */
+- (void)_synchronizeAllSettings
+{
+  [self _hotLoadingSettingDidChange];
+  [self _liveReloadSettingDidChange];
+  [self _remoteDebugSettingDidChange];
+  [self _profilingSettingDidChange];
+}
+
+- (void)_pollForLiveReload
+{
+  if (!_isJSLoaded || ![[self settingForKey:kRCTDevSettingLiveReloadEnabled] boolValue] || !_liveReloadURL) {
+    return;
+  }
+  
+  if (_liveReloadUpdateTask) {
+    return;
+  }
+  
+  __weak RCTDevSettings *weakSelf = self;
+  _liveReloadUpdateTask = [[NSURLSession sharedSession] dataTaskWithURL:_liveReloadURL completionHandler:
+                           ^(__unused NSData *data, NSURLResponse *response, NSError *error) {
+                             
+                             dispatch_async(dispatch_get_main_queue(), ^{
+                               __strong RCTDevSettings *strongSelf = weakSelf;
+                               if (strongSelf && [[strongSelf settingForKey:kRCTDevSettingLiveReloadEnabled] boolValue]) {
+                                 NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
+                                 if (!error && HTTPResponse.statusCode == 205) {
+                                   [strongSelf reload];
+                                 } else {
+                                   if (error.code != NSURLErrorCancelled) {
+                                     strongSelf->_liveReloadUpdateTask = nil;
+                                     [strongSelf _pollForLiveReload];
+                                   }
+                                 }
+                               }
+                             });
+                             
+                           }];
+  
+  [_liveReloadUpdateTask resume];
+}
+
+- (void)jsLoaded:(NSNotification *)notification
+{
+  if (notification.userInfo[@"bridge"] != _bridge) {
+    return;
+  }
+  
+  _isJSLoaded = YES;
+  
+  // Check if live reloading is available
+  NSURL *scriptURL = _bridge.bundleURL;
+  if (![scriptURL isFileURL]) {
+    // Live reloading is disabled when running from bundled JS file
+    _liveReloadURL = [[NSURL alloc] initWithString:@"/onchange" relativeToURL:scriptURL];
+  } else {
+    _liveReloadURL = nil;
+  }
+  
+  dispatch_async(dispatch_get_main_queue(), ^{
+    // update state again after the bridge has finished loading
+    [self _synchronizeAllSettings];
+  });
+}
+
+#pragma mark - RCTWebSocketObserver
```

Comment:
This is probably fine for now, but I think this should go in a separate class as well.